### PR TITLE
[Merged by Bors] - feat: add 2 `coe_pow` lemmas

### PR DESCRIPTION
**Pull request by urkud**
Also drop `@[simp]` on `LinearMap.pow_apply`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/NormedSpace/LinearIsometry.lean [hunk 1]
```diff
@@ -439,6 +439,10 @@ theorem mul_def (f g : E →ₗᵢ[R] E) : (f * g : E →ₗᵢ[R] E) = f.comp g
   rfl
 #align linear_isometry.mul_def LinearIsometry.mul_def
 
+@[simp]
```
Comment (urkud) [inline Mathlib/Analysis/NormedSpace/LinearIsometry.lean:442]:
```suggestion
```

File: Mathlib/Topology/Algebra/Module/Basic.lean [hunk 2]
```diff
@@ -860,6 +860,10 @@ instance monoidWithZero : MonoidWithZero (M₁ →L[R₁] M₁) where
   mul_assoc _ _ _ := ext fun _ => rfl
 #align continuous_linear_map.monoid_with_zero ContinuousLinearMap.monoidWithZero
 
+@[simp]
```
Comment (urkud) [inline Mathlib/Topology/Algebra/Module/Basic.lean:863]:
```suggestion
```

File: Mathlib/LinearAlgebra/Basic.lean [hunk 3]
```diff
@@ -321,11 +321,8 @@ theorem coeFn_sum {ι : Type _} (t : Finset ι) (f : ι → M →ₛₗ[σ₁₂
 #align linear_map.coe_fn_sum LinearMap.coeFn_sum
 
 @[simp]
-theorem pow_apply (f : M →ₗ[R] M) (n : ℕ) (m : M) : (f ^ n) m = f^[n] m := by
-  induction' n with n ih
-  · rfl
-  · simp only [Function.comp_apply, Function.iterate_succ, LinearMap.mul_apply, pow_succ, ih]
-    exact (Function.Commute.iterate_self _ _ m).symm
+theorem pow_apply (f : M →ₗ[R] M) (n : ℕ) (m : M) : (f ^ n) m = f^[n] m :=
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/Basic.lean:324]: Can you add a `coe_pow` lemma here too?

**Review by eric-wieser — APPROVED** [review]
bors d+

File: Mathlib/LinearAlgebra/Basic.lean (hunk 3 above)
Comment (urkud) [reply Mathlib/LinearAlgebra/Basic.lean]: It was there several lines below. I'm trying to drop `@[simp]` so that `LinearMap`s align with other types.

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**